pkg/operator/controller.go

```diff
@@ -179,6 +238,13 @@ func (c *Controller) sync() error {
 		klog.Warningf("unknown custom resource state: %s", cr.Spec.ManagementState)
 	}
 
+	// XXX
+	// if hashes are different we need to die.
+	if c.caBundleHash != remoteBundleHash {
```

Comment:
rather than calculating local/remote hashes you should be able to just watch the local file for changes.  if it changes, die (the operator will be restarted).  i believe some of the other operators already implement this pattern.

Reply:
(you do need to ensure changes to the configmap drive a rollout of the registry operand, but there is already a framework/logic for telling this operator to "watch all these configmaps and compute a hash from them.  put that hash on the operand deployment, so that if the hash ever changes, a deployment will be triggered").  you just need to make sure this new CA configmap is part of that list.

Reply:
@bparees I used the canonical example you provided below to re-implement this PR. Now we have this other container running as a sidecar that monitors for file changes. I had to update a few dependencies in order to import the `watchdog` package(hence the huge PR) and I also had to use a different binary name for the `watch` process as it was, sometimes, killing itself.

Reply:
@bparees I'm taking care of the "bounce the registry" logic in #349.